File: Q099/Q099.py
from math import log10
import os

def large_exponential(base, expo):
    return expo*log10(base)

def main():
    maximum = temp = count = 0
    with open(os.path.join(os.path.dirname(__file__), "base_exp.txt")) as datafile:
        for row in datafile:
            count += 1
            base, expo = (int(k) for k in row.split(','))
            result = large_exponential(base, expo)
            if temp < result:
                temp = result
                maximum = count
    print "The maximum exponential is in line ", maximum

if __name__ =="__main__":
    main()

File: Q099/Q099.rb
result = 1
linenum = largest = 0
IO.foreach "base_exp.txt" do |line|
  linenum += 1
  base, expo = line.split(',').collect{|i| i.to_i}
  benchmark = expo*Math.log10(base)
  if largest < benchmark then largest=benchmark; result = linenum end
end
puts result
